### PR TITLE
Getting started with Alert Dialog in Android-Kotlin

### LINKED ISSUE #1216: Introduction to Dialogs in Android using Kotlin
### Brief Summary:
 
Android has distinguished techniques for improving user experience. One of them is the use of dialogs. A dialog is a brief conversation between the user and the application. They can be used to request actions from the users or pass quick messages to them.

### Key Takeaways: 

By the end of this tutorial, the learner should be able to:

1. Understand what are Dialogs and their roles in an application
2. Create an alert Dialog using Kotlin
3. Design and implement different types of custom Dialogs

### References: 

N/A

### PR DESCRIPTION
**Pull request by Ericgacoki**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This Fixes #1216

### CODE REVIEW

File: articles/getting-started-with-dialogs-in-android-kotlin/index.md [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+<resources>
+    <string name="show_dialog">Show dialog</string>
+    <string name="show_custom_dialog">Show custom dialog</string>
+</resources>
+```
+*buttons preview*
+
+![buttons image](/engineering-education/getting-started-with-dialogs-in-android-kotlin/buttons.png)
+
+As mentioned earlier, we will use these buttons to show the necessary dialogs. To achieve this, we need DataBinding objects for each view in our layout. The beauty of DataBinding is that it generates the objects for us. We just need to enclose the layout with a `layout` tag as shown below.
```
Comment (kayere) [inline articles/getting-started-with-dialogs-in-android-kotlin/index.md:96]: I don't think this is necessary. Constraint layout is also a layout. So, the binding class is also generated without adding the layout tag.

File: articles/getting-started-with-dialogs-in-android-kotlin/index.md [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+}
+```
+
+### Creating Alert Dialogs
+An Alert Dialog is created by instantiating the `AlertDialog` class and then creating a `Builder` for it. There are several types of Alert Dialogs in Android. In this tutorial, we'll create every dialog in its own `function` and the respective one will be called when a button is clicked.
+
+#### A). Default Alert Dialog
+This type of dialog is usually rectangular and appears at the center of the screen. We'll discuss its functionalities as we proceed. Meanwhile, paste the code below inside the `MainActivity` class, just below the `onCreate()` method.
+
+```java
```
Comment (kayere) [inline articles/getting-started-with-dialogs-in-android-kotlin/index.md:142]: Use the correct language for the snippets.

File: articles/getting-started-with-dialogs-in-android-kotlin/index.md [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+```
+Also, copy and paste the code below just above or below the `showDefaultDialog()` function.
+
+```java
+    private fun toast(text: String) = Toast.makeText(this, text, Toast.LENGTH_SHORT).show()
+```
+
+**Explanation**
+
+The syntax for creating a button requires us to pass in a listener of type `DialogInterface.OnClickListener` in the lambda function. A `DialogInterface` defines a dialog-type class that can be shown, canceled or dismissed, and may have buttons that can be clicked. `Kotlin` simplifies this by allowing us to pass in `underscores` as arguments of the lambda parameters.
```
Comment (kayere) [inline articles/getting-started-with-dialogs-in-android-kotlin/index.md:208]: We use underscores when the parameter is not going to be used in the lambda function. Please incorporate that.

**Review by kayere — COMMENTED** [review]
Hello @Ericgacoki - Great article. Please go over the few suggestions I have made and the article will be good to go.

**Comment by Ericgacoki** [comment]
Hey @kayere,  I've made the suggested changes. Please review. Thank you.

**Review by kayere — APPROVED** [review]
Great article @Ericgacoki. 
@hectorkambow - Please give this article a final review.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
Great work gentlemen @Ericgacoki  

cc @kayere  🚀

**Comment by Ericgacoki** [comment]
Thank you @hectorkambow and @kayere.

**Pull request merged**